### PR TITLE
Fix tests of /initialSync to work when the response is being cached

### PR DESCRIPTION
**Pull request by NegativeMjark**
Work around the changes in matrix-org/synapse#457

### CODE REVIEW

File: tests/10apidoc/30room-create.pl [hunk 1]
```diff
@@ -26,13 +26,9 @@
...
          Future->done(1);
-      });
-   },
-
-   check => sub {
-      my ( $user ) = @_;
-
-      matrix_initialsync( $user )->then( sub {
+      })->then( sub {
+         matrix_initialsync( $user );
```
Comment (leonerd) [inline tests/10apidoc/30room-create.pl]: Rather that this, use the same `limit` trick as before. At the top of the file put something like

    # An incrementing parameter for initialSync to defeat the caching mechanism and ensure fresh results every time
    my $initial_sync_limit = 1;

Then change every occurance of `matrix_initialsync( $user )` to

    matrix_initialsync( $user, limit => $initial_sync_limit++ );

File: tests/30rooms/21receipts.pl [hunk 2]
```diff
@@ -48,7 +48,7 @@ sub find_receipt
 
          matrix_advance_room_receipt( $user, $room_id, "m.read" => $member_event_id )
       })->then( sub {
-         matrix_initialsync( $user )
+         matrix_initialsync( $user, limit => 1)
```
Comment (leonerd) [inline tests/30rooms/21receipts.pl]: See comment in earlier file for a neater idea.

Or keep this but fix the whitespace - `limit => 1 )`

File: tests/10apidoc/30room-create.pl [hunk 3]
```diff
@@ -32,7 +34,7 @@
    check => sub {
       my ( $user ) = @_;
 
-      matrix_initialsync( $user )->then( sub {
+      matrix_initialsync( $user, limit => $initial_sync_limit++ )->then( sub {
```
Comment (richvdh) [inline tests/10apidoc/30room-create.pl]: could you add a comment to say what the limit is for?

File: tests/30rooms/21receipts.pl [hunk 4]
```diff
@@ -48,7 +51,7 @@ sub find_receipt
 
          matrix_advance_room_receipt( $user, $room_id, "m.read" => $member_event_id )
       })->then( sub {
-         matrix_initialsync( $user )
+         matrix_initialsync( $user, limit => $initial_sync_limit++ )
```
Comment (richvdh) [inline tests/30rooms/21receipts.pl]: why do the others get a comment but not this one? could you add it here too

File: tests/30rooms/21receipts.pl [hunk 5]
```diff
@@ -96,7 +99,7 @@ sub find_receipt
          # Now lets check that they are monotonically racheted
          matrix_advance_room_receipt( $user, $room_id, "m.read" => $member_event_id );
       })->then( sub {
-         matrix_initialsync( $user );
+         matrix_initialsync( $user , limit => $initial_sync_limit++ );  # Change the limit to defeat caching
```
Comment (richvdh) [inline tests/30rooms/21receipts.pl]: nit: s/ ,/,/

**Comment by richvdh** [comment]
Leo's batshit whitespace conventions always catch me out, so I've probably missed something, but looks fine except for some nits

File: tests/10apidoc/30room-create.pl (hunk 3 above)
Comment (NegativeMjark) [inline tests/10apidoc/30room-create.pl]: Done 897709a

File: tests/30rooms/21receipts.pl (hunk 4 above)
Comment (NegativeMjark) [inline tests/30rooms/21receipts.pl]: Done 897709a

File: tests/30rooms/21receipts.pl (hunk 5 above)
Comment (NegativeMjark) [inline tests/30rooms/21receipts.pl]: Done 897709a

**Comment by leonerd** [comment]
Posthoc LGTM

**Pull request merged**